rest-tests/src/test/groovy/org/hawkular/metrics/rest/CORSITest.groovy

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2014-2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.rest
+
+import static org.junit.Assert.assertEquals
+
+import static org.hawkular.metrics.api.jaxrs.util.Headers.ACCESS_CONTROL_ALLOW_CREDENTIALS;
+import static org.hawkular.metrics.api.jaxrs.util.Headers.ACCESS_CONTROL_ALLOW_HEADERS;
+import static org.hawkular.metrics.api.jaxrs.util.Headers.ACCESS_CONTROL_ALLOW_METHODS;
+import static org.hawkular.metrics.api.jaxrs.util.Headers.ACCESS_CONTROL_ALLOW_ORIGIN;
+import static org.hawkular.metrics.api.jaxrs.util.Headers.ACCESS_CONTROL_MAX_AGE;
+import static org.hawkular.metrics.api.jaxrs.util.Headers.ACCESS_CONTROL_REQUEST_METHOD;
+import static org.hawkular.metrics.api.jaxrs.util.Headers.DEFAULT_CORS_ACCESS_CONTROL_ALLOW_HEADERS;
+import static org.hawkular.metrics.api.jaxrs.util.Headers.DEFAULT_CORS_ACCESS_CONTROL_ALLOW_METHODS;
+import static org.hawkular.metrics.api.jaxrs.util.Headers.ORIGIN;
+
+import org.junit.Test
+
+class CORSITest extends RESTTest {
+
+    @Test
+    void testOptionsWithOrigin() {
+        def testOrigin = "TestOrigin123";
+
+        def response = hawkularMetrics.options(path: "ping",
+         headers: [
+            (ACCESS_CONTROL_REQUEST_METHOD): "POST",
+            (ORIGIN): testOrigin,
+         ]);
+
+         //print headers
+         println("==== Request Headers = Start  ====");
```

Comment:
Could you remove this? We've tried to avoid useless output so far. If you need to print it in case of an error, then you can build a string with all the headers, and add it as 1st param of assertXXX calls.